students/gerry_gabrisch/lesson03/mailroom.py

```diff
@@ -0,0 +1,121 @@
+        #Sort the list based on total donations...
+        reportList.append(thisUsersReport)
+    #call the sorting function...
+    reportList = SortList(reportList)
+    #call the report format function...
+    ReportFormatAndPrint(reportList)
+
+def ReportFormatAndPrint(reportList):
+    '''takes the sorted report and formats and prints the report to the screen... '''
+    #print the report header
+    print('\nDonor Name                | Total Given | Num Gifts | Average Gift')
+    #for every donator get the donation stats, format and print them...
+    for item in reportList:
+        print(f'{item[0]:26} ${item[1]:>12.2f} {item[2]:>11}  ${item[3]:>11.2f}')
+    print()
+    
+def SortList(reportList):
+    '''sorts the list using itemgetter from highest donor to lowest..'''
+    from operator import itemgetter
+    #sorting sorts from lowest to highest unless you use the reverse option...
+    reportList = sorted(reportList, key=itemgetter(1), reverse=True)
+    return reportList
+
+def MakeDonorList(donations):
+    '''this just makes a new list of donor names so I don't have to write multiple iterators...'''
+    donorsOnly = []
+    for i in donations:
+        donorsOnly.append((i[0]))  
+    return donorsOnly
+
+def ListDonors(list):
+    '''prints the donor's names using this cool \n.join trick...'''
+    print('\n'+'\n'.join(list)+'\n')
+
+def MakeNewDonor(donations, donor_name):
+    '''make a new donor record if the user name is not an existing donor...'''
+    #make a new list to hold the new donor and append the name to this list...
+    new_donor = []
+    #add the donor to element position 0...
+    new_donor.append(donor_name)
+    #now append this list to hold the donations...
+    new_donor.append([])
+    #add the new donor record to the list of all donations...
+    donations.append(new_donor)
+    return donations
+    
+def TakeDonation(donations, donor_name): 
+    '''gets a new donation amount and calls the email formatter...'''
+    donationAmount = float(input('Please enter a donation amount >'))
+    #find the donor and add this donation to their list of donations...    
+    for i in donations:
+        if i[0] == donor_name:
+            i[1].append(donationAmount)
+            #call the mail format function...
+            FormatMailer(i)
+    
+
+def FormatMailer(i):
+    '''print the thank you email to the screen....'''
+    print()
+    print(f'Dear {i[0]},\nThank you.  The Ministry of Silly Walks appreciates your donation of ${i[1][-1]:.2f}.\nRespectfully\nGerry\nGerry@MinistryofSillyWalks.com')
+    print()
+
+
+def main():
+    while True:
+        promp_response = input('Would you like to \n1: Take a donation and send a Thank You \n2: Create a Report or \n3: Quit? Enter a number now.>')
+        if promp_response == '1':  
+            SendThanks(donations)
+        if promp_response == '2':
+            CreateReportList(donations)
```

Comment:
no need to pass in the global donations structure (this will make more difference in lesson04)